src/index.ts

```diff
@@ -5,16 +5,10 @@ import { restEndpointMethods } from "@octokit/plugin-rest-endpoint-methods";
 
 import { VERSION } from "./version";
 
-export const Octokit = Core
-  // TODO: this really should be
-  //
-  //     .plugin([requestLog, paginateRest, restEndpointMethods])
-  //
-  // but for mystical reasons, using the line above does set the resulting the
-  // `octokit` instance type correctly. Neither `octokit.paginate()` nor all the
-  // endpoint methods such as `octokit.repos.get() are set
-  .plugin(requestLog)
-  .plugin([paginateRest, restEndpointMethods])
-  .defaults({
-    userAgent: `octokit-rest.js/${VERSION}`
-  });
+export const Octokit = Core.plugin([
+  (requestLog as unknown) as () => void,
```

Comment:
Oh this is a much more concise workaround, I like it! Could you maybe add a comment to explain what exactly is happening?

We should probably document the workaround for TypeScript users in https://github.com/octokit/rest.js/blob/master/docs/src/pages/api/08_plugins.md, what do you think?

Reply:
> We should probably document the workaround for TypeScript users

I think we should come up with a solution for  octokit/core.js#51 before updating the documentation. Depending on the solution, different documentation changes will be made for both plugin authors and plugin consumers.